api/controllers/CampaignBotController.js

```diff
@@ -1,548 +1,454 @@
 "use strict";
 
+/**
+ * Imports.
+ */
 const logger = rootRequire('lib/logger');
 const mobilecommons = rootRequire('lib/mobilecommons');
 const phoenix = rootRequire('lib/phoenix')();
 const helpers = rootRequire('lib/helpers');
-
 const connOps = rootRequire('config/connectionOperations');
 const dbRbSubmissions = require('../models/campaign/ReportbackSubmission')(connOps);
 const dbSignups = require('../models/campaign/Signup')(connOps);
 const dbUsers = require('../models/User')(connOps);
 
+/**
+ * Setup.
+ */
 const CMD_REPORTBACK = (process.env.GAMBIT_CMD_REPORTBACK || 'P');
 
 /**
- * CampaignBotController
- * @constructor
- * @param {integer} campaignId - our DS Campaign ID
+ * CampaignBotController.
  */
-function CampaignBotController(campaignId) {
-  this.campaignId = campaignId;
-  this.campaign = app.getConfig(app.ConfigName.CAMPAIGNS, campaignId);
-  if (!this.campaign) {
-    return;
-  }
-
-  // @todo Config variable for default CampaignBot ID
-  // @todo Store campaignbot property on our config.campaigns to override bot
-  // on a per DS Campaign basis.
-  this.bot = app.getConfig(app.ConfigName.CAMPAIGNBOTS, 41);
+class CampaignBotController {
 
-  var mobileCommonsCampaign = this.campaign.staging_mobilecommons_campaign;
+  constructor(req, res) {
```

Comment:
Indeed. moving all of the routing logic like
- if we don't have a cached user, create a user, then check for cached signup, if no signup check Phoenix for a Signup, if no Signup, post to Phoenix and cache new signup
- if we have a user, signup, and draft reportback submission, and don't have quantity yet, ask for quantity
- etc

to someplace like `router.js` def makes sense, and our `CampaignBotController` just has functions that create the cache documents, get/post to API etc.

Solid JS'ing @deadlybutter -- thanks for the feedback!